packages/cardano-services/test/Asset/TypeormAssetProvider/TypeormAssetProvider.test.ts

```diff
@@ -0,0 +1,118 @@
+import { clearDbPools, sleep } from '../../util';
+import { logger } from '@cardano-sdk/util-dev';
+
+describe('TypeormAssetProvider', () => {
+  let provider: TypeormAssetProvider;
+  let service: TypeOrmNftMetadataService;
+  let fixtureBuilder: TypeormAssetFixtureBuilder;
+  let dbPools: DbPools;
+
+  beforeAll(async () => {
+    const dnsResolver = createDnsResolver({ factor: 1.1, maxRetryTime: 1000 }, logger);
+    const entities = getEntities(['asset']);
+    const connectionConfig$ = getConnectionConfig(dnsResolver, 'test', 'Handle', {
+      postgresConnectionStringHandle: process.env.POSTGRES_CONNECTION_STRING_HANDLE!
+    });
+
+    service = new TypeOrmNftMetadataService({ connectionConfig$, entities, logger });
+    provider = new TypeormAssetProvider({ connectionConfig$, entities, logger, nftMetadataService: service });
+    await provider.initialize();
+    await provider.start();
+    await service.initialize();
+    await service.start();
+  });
+
+  afterAll(async () => {
+    jest.restoreAllMocks();
+
+    await provider.shutdown();
+    await service.shutdown();
+  });
+
+  beforeEach(async () => {
+    dbPools = {
+      healthCheck: new Pool({ connectionString: process.env.POSTGRES_CONNECTION_STRING_HANDLE }),
+      main: new Pool({ connectionString: process.env.POSTGRES_CONNECTION_STRING_HANDLE })
+    };
+    fixtureBuilder = new TypeormAssetFixtureBuilder(dbPools.main, logger);
+  });
+
+  afterEach(async () => {
+    await clearDbPools(dbPools);
+  });
+
+  describe('healthCheck', () => {
+    it('forwards the assetProvider health response with provider client', async () => {
+      // required a delay to determine TypeormProvider's healthCheck as healthy when subscribes to observable data source
+      const response = await provider.healthCheck();
+      expect(response).toEqual({ ok: false, reason: 'not started' });
+      while (!(await provider.healthCheck()).ok) await sleep(10);
+      const response2 = await provider.healthCheck();
+      expect(response2).toEqual({ ok: true });
+    });
+  });
+
+  describe('getAsset', () => {
+    it('should return asset info for a given asset id', async () => {
+      const assetQuery = await fixtureBuilder.getAssets(1);
+      const testAsset = assetQuery[0];
+      const asset = await provider.getAsset({ assetId: testAsset.assetId });
+      expect(asset).toEqual(testAsset);
+    });
+
+    it('should return throw an error if the asset does not exist', async () => {
+      const assetId = Cardano.AssetId('64190c10b567ad4bcf254ad0f76eb374d749d0b25fd82786af6a839a48656c6c6f48616e646c65');
+      await expect(provider.getAsset({ assetId })).rejects.toThrowError(ProviderError);
+    });
+
+    it('should return metadata if the asset has metadata', async () => {
+      const assetQuery = await fixtureBuilder.getAssets(1, { with: [TypeormAssetWith.metadata] });
+      const testAsset = assetQuery[0];
```

Comment:
Looking at fixture builder code, it doesn't look like the returned assets are guaranteed to have nft metadata: `Queries.findAssets` returns any (with or without metadata) assets
```suggestion
      const testAsset = assetQuery[0];
      expect(typeof testAsset.nftMetadata).toBe('object');
```

Reply:
Yes. It only returns `nftMetadata` when it's called like this.

```
await fixtureBuilder.getAssets(1, { hasNoMetadata: true, with: [TypeormAssetWith.metadata] })
```

Reply:
Based on property name has**No**Metadata it should do the opposite :thinking:

Reply:
`hasNoMetadata` only fetches assetInfo for assets that have no metadata, it does not query the metadata table

`[TypeormAssetWith.metadata]` queries the `nft_metadata` table.

Reply:
Right, but not all assets have nft metadata

Reply:
Fixed on 41f204f